src/main/java/org/broadleafcommerce/menu/admin/server/handler/MenuItemCustomPersistenceHandler.java

```diff
@@ -218,8 +234,67 @@ protected FilterMapping createFilterMappingForProperty(String targetPropertyName
         Restriction newRestriction = new Restriction().withPredicateProvider(predicateProvider);
 
         return new FilterMapping()
-            .withFieldPath(fieldPath)
-            .withDirectFilterValues(directFilterValues)
-            .withRestriction(newRestriction);
+                .withFieldPath(fieldPath)
+                .withDirectFilterValues(directFilterValues)
+                .withRestriction(newRestriction);
+    }
+
+    protected Entity validateMenuItem(final Entity entity) throws ServiceException {
+        final Entity entityAfterValidateRecursiveRelationship = this.validateRecursiveRelationship(entity);
+        return this.validateCategory(entityAfterValidateRecursiveRelationship);
     }
+
+    protected Entity validateCategory(Entity entity) {
+        final String currentCategoryId = entity.getPMap().get("id").getValue();
+        final String defaultCategoryId = entity.getPMap().get("linkedMenu").getValue();
+        if (currentCategoryId != null && currentCategoryId.equals(defaultCategoryId)) {
+            entity.addValidationError(LINKED_MENU_PROPERTY, "validationCategory");
+        }
+        return entity;
+    }
+
+    protected Entity validateRecursiveRelationship(Entity entity) throws ServiceException {
+        try {
+            String linkedMenuId = entity.findProperty(LINKED_MENU_PROPERTY).getValue();
```

Comment:
I think we can Parse "long" only here and in the following methods use just Long